packages/react-vapor/src/components/textarea/TextArea.tsx

```diff
@@ -57,43 +57,60 @@ const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITextAreaOwnProps): I
     onUnmount: () => dispatch(removeTextArea(ownProps.id)),
 });
 
-export class TextArea extends React.Component<ITextAreaProps, {}> {
-    static defaultProps: Partial<ITextAreaOwnProps> = {
-        additionalAttributes: {},
-        className: '',
-    };
+export const TextArea: React.FunctionComponent<ITextAreaProps> = (props) => {
+    const [debouncedValue, setDebouncedValue] = React.useState(props.value);
+    const [isValid, setIsValid] = React.useState(true);
 
-    componentWillMount() {
-        if (this.props.onMount) {
-            this.props.onMount();
-        }
-    }
+    React.useEffect(() => {
+        const timeout = setTimeout(() => {
+            setDebouncedValue(props.value);
+        }, 300);
+        return () => clearTimeout(timeout);
+    }, [props.value]);
 
-    componentWillUnmount() {
-        if (this.props.onUnmount) {
-            this.props.onUnmount();
-        }
-    }
+    React.useEffect(() => {
+        setIsValid(props.validate?.(debouncedValue));
+    }, [debouncedValue]);
 
-    render() {
-        const TextareaTagName: any = this.props.isAutosize ? TextareaAutosize : 'textarea';
+    React.useEffect(() => {
+        props.onMount?.();
+        setIsValid(true);
+        return props.onUnmount;
+    }, []);
+
+    const getValidationLabel = () => {
         return (
-            <TextareaTagName
-                {...this.props.additionalAttributes}
-                id={this.props.id}
-                disabled={this.props.disabled}
-                className={this.props.className}
-                value={this.props.value}
-                onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => this.handleOnChange(e)}
-            />
+            !isValid && (
+                <div className={'pt1'}>
+                    <Label id={props.id} className="text-red" {...props.validationLabelProps}>
+                        {props.validationMessage}
+                    </Label>

```

Comment:
I'm wondering if we really need a Label here, we have a class that adds all the correct style, see https://github.com/coveo/react-vapor/blob/f48a6b08481988daf378598693be352441d73dea/packages/react-vapor/src/components/numericInput/NumericInputConnected.tsx#L116 for an exemple